### PR TITLE
[Bug 20405] Fixed bug causing incorrect formattedWidth to be returned

### PR DESCRIPTION
**Pull request by abrisan**

### CODE REVIEW

File: engine/src/exec-interface-button.cpp [hunk 1]
```diff
@@ -924,23 +924,39 @@ void MCButton::SetMnemonic(MCExecContext& ctxt, uinteger_t p_mnemonic)
...
 				fwidth = 0;
-			else 
-                fwidth = leftmargin + rightmargin + MCFontMeasureText(m_font, t_label, getstack() -> getdevicetransform());
-			
+			else
+            {
+                MCArrayRef lines;
+                MCStringSplit(t_label, MCSTR("\n"), nil, kMCCompareExact, lines);
+                uindex_t line_count = MCArrayGetCount(lines);
+                uindex_t max_length = 0;
```
Comment (runrevmark) [inline engine/src/exec-interface-button.cpp]: This should match the return type of MCFontMeasureText which is int32_t

File: engine/src/exec-interface-button.cpp [hunk 2]
```diff
@@ -924,23 +924,39 @@ void MCButton::SetMnemonic(MCExecContext& ctxt, uinteger_t p_mnemonic)
...
+                
+                for (uindex_t i = 0 ; i < line_count ; ++i)
```
Comment (runrevmark) [inline engine/src/exec-interface-button.cpp]: We usually don't put spaces before ;

File: engine/src/exec-interface-button.cpp [hunk 3]
```diff
@@ -924,23 +924,39 @@ void MCButton::SetMnemonic(MCExecContext& ctxt, uinteger_t p_mnemonic)
...
+                {
+                    MCValueRef line = nil;
+                    MCArrayFetchValueAtIndex(lines, i+1, line);
```
Comment (runrevmark) [inline engine/src/exec-interface-button.cpp]: Put spaces around binary operators

**Comment by runrevmark** [comment]
@livecode-vulcan review ok cad6272

**Comment by livecode-vulcan** [comment]
💙 review by @runrevmark ok cad6272eb67d292fa1153bb9510eaf672915843b

**Comment by livecode-vulcan** [comment]
😞 test failure cad6272eb67d292fa1153bb9510eaf672915843b

* try-community-armv6-android-api8: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api8/builds/818)
* try-community-armv6-android-api9: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api9/builds/834)
* try-community-js-emscripten-sdk1.35: [success](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/808)
* try-community-universal-ios-iphoneos10.3: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos10.3/builds/752)
* try-community-universal-ios-iphonesimulator10.3: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator10.3/builds/755)
* try-community-universal-mac-macosx10.6: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.6/builds/812)
* try-community-universal-mac-macosx10.9: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/834)
* try-community-x86-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian7/builds/817)
* try-community-x86-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/834)
* try-community-x86_64-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian7/builds/816)
* try-community-x86_64-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/827)
* try-community-x86-win32: [failure](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/848)
* try-community-x86_64-win32: [pending](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/813)

**Comment by livecodeali** [comment]
@livecode-vulcan retry cad6272

**Comment by livecode-vulcan** [comment]
😎 test success cad6272eb67d292fa1153bb9510eaf672915843b

* try-community-armv6-android-api8: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api8/builds/825)
* try-community-armv6-android-api9: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api9/builds/841)
* try-community-js-emscripten-sdk1.35: [success](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/815)
* try-community-universal-ios-iphoneos10.3: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos10.3/builds/759)
* try-community-universal-ios-iphonesimulator10.3: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator10.3/builds/762)
* try-community-universal-mac-macosx10.6: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.6/builds/819)
* try-community-universal-mac-macosx10.9: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/841)
* try-community-x86-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian7/builds/824)
* try-community-x86-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/841)
* try-community-x86_64-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian7/builds/823)
* try-community-x86_64-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/834)
* try-community-x86-win32: [success](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/855)
* try-community-x86_64-win32: [success](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/820)

**Pull request merged**